sway/desktop/xwayland.c

```diff
@@ -234,10 +283,18 @@ static void handle_commit(struct wl_listener *listener, void *data) {
 	struct sway_xwayland_view *xwayland_view =
 		wl_container_of(listener, xwayland_view, commit);
 	struct sway_view *view = &xwayland_view->view;
-	// NOTE: We intentionally discard the view's desired width here
-	// TODO: Let floating views do whatever
-	view_update_size(view, xwayland_view->pending_width,
-		xwayland_view->pending_height);
+	struct wlr_xwayland_surface *xsurface = view->wlr_xwayland_surface;
+	if (!view->natural_width && !view->natural_height) {
+		view->natural_width = xsurface->width;
+		view->natural_height = xsurface->height;
+	}
+	if (view->swayc && container_is_floating(view->swayc)) {
+		view_update_size(view, xsurface->width, xsurface->height);
```

Comment:
Not sure we want to allow xwayland windows to move. We want to allow that for unmanaged windows, sure, but do we want to do this for toplevels?

Reply:
This is resolved in commit "Don't let xwayland views set position unless unmanaged".